src/inputs/context-menu/PContextMenu.vue

```diff
@@ -175,200 +172,170 @@ const getFilteredSelectedItems = (selected: MenuItem[], menu: MenuItem[]): MenuI
-        },
-        noSelectIndication: {
-            type: Boolean,
-            default: false,
-        },
-        showSelectHeader: {
-            type: Boolean,
-            default: false,
-        },
-        showClearSelection: {
-            type: Boolean,
-            default: false,
-        },
-        searchable: {
-            type: Boolean,
-            default: false,
-        },
-        searchText: {
-            type: String,
-            default: '',
-        },
-    },
-    setup(props, { emit, slots }) {
-        const state = reactive({
-            proxySearchText: props.searchText ?? '',
-            isFocusedOnSearch: false,
-            proxySelected: useProxyValue<MenuItem[]>('selected', props, emit),
-            selectedNameMap: computed<Record<string, number>>(() => {
-                const selectedMap = {};
-                state.proxySelected.forEach((item, idx) => {
-                    selectedMap[item.name] = idx;
-                });
-                return selectedMap;
-            }),
-            selectableMenuItems: computed(() => props.menu.filter((d) => !d.disabled && (d.type === undefined || d.type === 'item'))),
-            selectedCount: computed(() => {
-                if (props.strictSelectMode) return state.selectableMenuItems.filter((d) => state.selectedNameMap[d.name] !== undefined).length;
-                return state.proxySelected.length;
-            }),
-            menuItemLength: computed(() => props.menu.filter((d) => d.type === undefined || d.type === 'item').length),
+interface ContextMenuProps {
+    menu: MenuItem[];
+    loading?: boolean;
+    selected?: MenuItem[];
+    multiSelectable?: boolean;
+    showRadioIcon?: boolean;
+    strictSelectMode?: boolean;
+    itemHeightFixed?: boolean;
+    highlightTerm?: string;
+    noSelectIndication?: boolean;
+    showSelectHeader?: boolean;
+    showClearSelection?: boolean;
+    searchable?: boolean;
+    searchText?: string;
+}
+interface ContextMenuEmits {
+    (e: 'update:selected', selected: MenuItem[]): void,
+    (e: 'focus', focusedIdx?: number): void,
+    (e: 'blur'): void,
+    (e: 'keyup:up:end'): void,
+    (e: 'keyup:down:end'): void,
+    (e: 'select', item: MenuItem, index: number): void,
+    (e: 'update:search-text', searchText: string): void,
+    (e: 'keyup:esc', mouseEvent: MouseEvent): void,
+    (e: 'click-button', item: MenuItem, index: number, mouseEvent: MouseEvent): void,
+    (e: 'click-done', mouseEvent: MouseEvent): void,
+    (e: 'click-show-more', item: MenuItem, index: number, mouseEvent: MouseEvent): void,
+}
+const props = withDefaults(defineProps<ContextMenuProps>(), {
+    menu: () => [],
+    loading: false,
+    selected: () => [],
+    multiSelectable: false,
+    showRadioIcon: false,
+    strictSelectMode: false,
+    itemHeightFixed: false,
+    highlightTerm: '',
+    noSelectIndication: false,
+    showSelectHeader: false,
+    showClearSelection: false,
+    searchable: false,
```

Comment:
question: Does `boolean` type also need a default value? Is it because of `undefined`?

Reply:
I removed it! thanx!